pool/paymentmgr_test.go

```diff
@@ -947,4 +993,709 @@ func testPaymentMgr(t *testing.T, db *bolt.DB) {
+	// cannot be published.
+	txBytes := []byte("01000000018e17619f0d627c2769ee3f957582691aea59c2" +
+		"e79cc45b8ba1f08485dd88d75c0300000001ffffffff017a64e43703000000" +
+		"00001976a914978fa305bd66f63f0de847338bb56ff65fa8e27288ac000000" +
+		"000000000001f46ce43703000000846c0700030000006b483045022100d668" +
+		"5812801db991b72e80863eba7058466dfebb4aba0af75ab47bade177325102" +
+		"205f466fc47435c1a177482e527ff0e76f3c2c613940b358e57f0f0d78d5f2" +
+		"ffcb012102d040a4c34ae65a2b87ea8e9df7413e6504e5f27c6bde019a78ee" +
+		"96145b27c517")
+	mgr.cfg.FetchTxBroadcaster = func() TxBroadcaster {
+		return &txBroadcasterImpl{
+			signTransaction: func(ctx context.Context, req *walletrpc.SignTransactionRequest, options ...grpc.CallOption) (*walletrpc.SignTransactionResponse, error) {
+				return &walletrpc.SignTransactionResponse{
+					Transaction: txBytes,
+				}, nil
+			},
+			publishTransaction: func(ctx context.Context, req *walletrpc.PublishTransactionRequest, options ...grpc.CallOption) (*walletrpc.PublishTransactionResponse, error) {
+				return nil, fmt.Errorf("unable to publish transaction")
+			},
+		}
+	}
+
+	err = mgr.payDividends(ctx, estMaturity+1, treasuryActive)
+	if !errors.Is(err, ErrPublishTx) {
+		cancel()
+		t.Fatalf("expected a publish error, got %v", err)
+	}
+
+	// Ensure paying dividend payment succeeds with valid inputs.
+	txHash, _ := hex.DecodeString("013264da8cc53f70022dc2b5654ebefc9ecfed24ea18dfcfc9adca5642d4fe66")
+	mgr.cfg.FetchTxBroadcaster = func() TxBroadcaster {
+		return &txBroadcasterImpl{
+			signTransaction: func(ctx context.Context, req *walletrpc.SignTransactionRequest, options ...grpc.CallOption) (*walletrpc.SignTransactionResponse, error) {
+				return &walletrpc.SignTransactionResponse{
+					Transaction: txBytes,
+				}, nil
+			},
+			publishTransaction: func(ctx context.Context, req *walletrpc.PublishTransactionRequest, options ...grpc.CallOption) (*walletrpc.PublishTransactionResponse, error) {
+				return &walletrpc.PublishTransactionResponse{
+					TransactionHash: txHash,
+				}, nil
+			},
+		}
+	}
+
+	err = mgr.payDividends(ctx, estMaturity+1, treasuryActive)
+	if err != nil {
+		cancel()
+		t.Fatalf("unexpected dividend payment error, got %v", err)
+	}
+
+	// Reset backed up values to their defaults.
+	mgr.setLastPaymentHeight(0)
+	mgr.setLastPaymentPaidOn(0)
+	mgr.setLastPaymentCreatedOn(0)
+	err = db.Update(func(tx *bolt.Tx) error {
+		err := mgr.persistLastPaymentHeight(tx)
+		if err != nil {
+			return fmt.Errorf("unable to persist default last "+
+				"payment height: %v", err)
+		}
+		err = mgr.persistLastPaymentPaidOn(tx)
+		if err != nil {
+			return fmt.Errorf("unable to persist default last "+
+				"payment paid on: %v", err)
+		}
+		err = mgr.persistLastPaymentCreatedOn(tx)
+		if err != nil {
+			return fmt.Errorf("unable to persist default last "+
+				"payment created on: %v", err)
+		}
+		return nil
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cancel()
```

Comment:
`cancel()` isnt called if an error is returned by any of the db calls above. Perhaps move this `cancel()` above `// Reset backed up values to their defaults.` ?